tests/phpunit/DataValueTest.php

```diff
@@ -70,29 +69,25 @@ function ( array $args ) use ( $instanceBuilder ) {
 	/**
 	 * @dataProvider validConstructorArgumentsProvider
 	 *
+	 * @param mixed ...$params
 	 * @since 0.1
 	 */
-	public function testConstructorWithValidArguments() {
-		$dataItem = call_user_func_array(
-			[ $this, 'newInstance' ],
-			func_get_args()
-		);
+	public function testConstructorWithValidArguments( ...$params ) {
```

Comment:
This is scaring me. Unfortunately this class is package public and derived from in other packages. I'm not sure if this is a breaking change or not and the tests here won't catch if it is.

Ideally we'd get rid of this test class. Removing it will cause a lot of work due to https://www.entropywins.wtf/blog/2018/08/30/base-libraries-should-be-stable/ though. So just slapping on a deprecation tag is probably better.

Reply:
I guess this can wait, then. I've rolled function signatures back.